test/integration/targets/nxos_install_os/tasks/network_cli.yaml

```diff
@@ -0,0 +1,16 @@
+---
+- name: collect common test cases
+  find:
+    paths: "{{ role_path }}/tests/common"
+    patterns: "{{ testcase }}.yaml"
+  connection: local
+  register: test_cases
+
+- name: set test_items
+  set_fact: test_items="{{ test_cases.files | map(attribute='path') | list }}"
+
+- name: run test cases (connection=network_cli)
+  include: "{{ test_case_to_run }} ansible_connection=network_cli connection={}"
```

Comment:
There's your problem, replace `connection={}` with `connection={{ cli }}` (or, again, anything really, as long as it's a dictionary) and #40027 will go away

Reply:
Thanks.  Will update.